libs/filamat/include/filamat/MaterialBuilder.h

```diff
@@ -805,17 +805,16 @@ class UTILS_PUBLIC MaterialBuilder : public MaterialBuilderBase {
     // Returns true if any of the parameter samplers matches the specified type.
     bool hasSamplerType(SamplerType samplerType) const noexcept;
 
-    static constexpr size_t MAX_PARAMETERS_COUNT = 48;
     static constexpr size_t MAX_SUBPASS_COUNT = 1;
     static constexpr size_t MAX_BUFFERS_COUNT = 4;
-    using ParameterList = Parameter[MAX_PARAMETERS_COUNT];
+    using ParameterList = std::vector<Parameter>;
     using SubpassList = Parameter[MAX_SUBPASS_COUNT];
     using BufferList = std::vector<std::unique_ptr<filament::BufferInterfaceBlock>>;
     using ConstantList = std::vector<Constant>;
     using PushConstantList = std::vector<PushConstant>;
 
     // returns the number of parameters declared in this material
-    uint8_t getParameterCount() const noexcept { return mParameterCount; }
+    uint8_t getParameterCount() const noexcept { return mParameters.size(); }
```

Comment:
potential for overflow here.  effectively limiting the size to 256.

Reply:
oh wow. good catch.